view/frontend/web/js/model/adyen-payment-service.js

```diff
@@ -146,11 +146,18 @@ define(
             },
 
             saveStateData: function (stateData) {
-                let serviceUrl = urlBuilder.createUrl('/adyen/carts/mine/state-data', {});
+                let serviceUrl;
+
+                if (customer.isLoggedIn()) {
+                    serviceUrl = urlBuilder.createUrl('/adyen/carts/mine/state-data', {});
+                } else {
+                    serviceUrl = urlBuilder.createUrl('/adyen/guest-carts/:cartId/state-data', {
+                        cartId: quote.getQuoteId()
+                    });
+                }
```

Comment:
could be more readable

```suggestion
                let urlPath = '/adyen/guest-carts/:cartId/state-data'
                let urlParams = {cartId: quote.getQuoteId()}
                
                if (customer.isLoggedIn()) {
                    urlPath = '/adyen/carts/mine/state-data';
                    urlParams = {}
                } 
                
                serviceUrl = urlBuilder.createUrl(urlPath, urlParams);
```

Reply:
Hello @hossam-adyen, thank you for you review. Let me implement it for other two function also.